util/pyclient/bft_msgs.py

```diff
@@ -65,11 +65,21 @@ def pack_request_header(header, pre_process=False):
     return b''.join([struct.pack(MSG_TYPE_FMT, msg_type),
                      struct.pack(REQUEST_HEADER_FMT, *header)])
 
-def unpack_request(data, cid_size = 0):
+def unpack_request(data):
     """Take a buffer and return a pair of the RequestHeader and app data"""
+    header = unpack_request_header(data)
+
     start = MSG_TYPE_SIZE + REQUEST_HEADER_SIZE
-    end = len(data) - cid_size
-    return unpack_request_header(data), data[start:end], data[end:].decode()
+    end = start + header.span_context_size
+    span_context = data[start:end]
+
+    start = end
+    end = start + header.length
+    msg = data[start:end]
+
+    start = end
+    cid = data[start:].decode()
+    return header, span_context, msg, cid
```

Comment:
This is a nit, but I find it inconsistent that the `span_context` gets returned before `msg, but `cid` after message. To me, they are really doing a very similar job, and are metadata after the header and message. I feel like they should either both be before message, or both after. Additionally, this should be reflected in how the messages themselves get constructed. If CID comes after the message in a packed struct, then why shouldn't the span_context ?

Reply:
The parts of the message are returned in the same order they're stored(header, span, msg_body, cid). The CID is actually stored after the message body and it is specific for only for client request.
I believe that it is preferable to behave like every other binary protocol:
header/payload, in our case everything that is not message body is header(even if it contains fields with variable length).
Besides, `cid` is supposed to be before the message body but I am afraid this is beyond the scope of this commit.

Reply:
I agree with your assessment. Thanks!